membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -89,6 +90,26 @@ class AttributeController(
       .map(maybeAttributes => ("supporter-product-data", maybeAttributes.getOrElse(None)))
   }
 
+  private def isLiveApp(ua: String): Boolean = ua.matches("""^Guardian(News)?\/.*""")
+  private def isHighContributor(contribution: ContributionAmount, isMonthly: Boolean): Boolean = {
+    val threshold = contribution.currency match {
+      case Currency.GBP | Currency.USD | Currency.EUR =>
+        if (isMonthly) 10 else 95
+      case Currency.CAD =>
+        if (isMonthly) 13 else 120
+      case Currency.AUD | Currency.NZD =>
+        if (isMonthly) 15 else 140
+    }
+    contribution.amount >= threshold
+  }
+  private def upgradeRecurringContributorsOnApps(userAgent: Option[String], attributes: Attributes): Attributes =
+    if (
+      userAgent.exists(isLiveApp) &&
+      attributes.RecurringContributionAmount.exists(amount => isHighContributor(amount, attributes.isMonthlyRecurringContributor))
+    ) {
+      attributes.copy(SupporterPlusExpiryDate = Some(LocalDate.now().plusDays(1)))
```

Comment:
Are we expecting the effect of this date to be that the apps will check daily? Do you know how that compares to the current check frequency?

Reply:
actually I'm not convinced this date matters at all. E.g. the android app [just checks a bool](https://github.com/guardian/android-news-app/blob/09f754ddb5872c0f6823f3262c03687358ae1647/android-news-app/src/main/java/com/guardian/feature/money/readerrevenue/SyncMembersDataApi.kt#L87) in the MDAPI response.

Only MDAPI cares about this date, when setting that bool: https://github.com/guardian/members-data-api/blob/main/membership-attribute-service/app/models/Attributes.scala#L53